_posts/2016-03-28-chime-quality-assurance.markdown

```diff
@@ -0,0 +1,47 @@
+author: "Nick Fagen"
+---
+![Sittercity QA](/assets/sittercity_qa_logo.png)
+
+So far, 2016 has been a great year. Our small, but rockstar, QA team has been making huge improvements towards improving automated regression suites across all of our projects. We have tackled the endless QA dilemma regarding automation, how to do it effectively and efficiently. There are always obstacles when automating any app. For example, let's say you have a new release and bug fixes ready to ship. How will you ensure that the new bug fix(es) have not introduced any new issues in previous working functionality? Manually testing all the functionality every single time you have some bug fixes or new feature addition is an option when a project is just starting.  However, as a project grows, large manual test regression plans are daunting and in my opinion a red flag. With larger regression plans you might be able to accomplish it, but then you are not going to be testing effectively in terms of company cost, resources, time, etc. More and more companies are trying to move toward continuous delivery, making a need for QA automation.
+
+As Bob V. stated in the last QA post, we have made a lot of changes. The last few years the QA department has accomplished a lot. What would take the QA team days to accomplish can now occur daily. In the past, if there were bugs, the release would have to wait until the next week because of the regression feedback loop. Now, for our newest product [Chime](https://hellochime.com) we deliver to prod multiple times a day, up to every 20 minutes for the varying underlying services. It is obviously not just QA's effort that has made this possible though; it has been a team effort across all departments (Development, Devops, and QA). Any company can accomplish continuous delivery if they have a determined team that can work well with one another.
+
+## Deciding what tools to use for Chime
+
+The newest Sittercity product, Chime, gave the QA team the opportunity to really work towards improving automation. There are many automation tools available. The foundational code a site uses will factor into what option is most appropriate. We primarily use ruby but also have a lot of different languages used in various projects. These may not work specifically for you, but they have met our needs so far: Watir, Cucumber, Appium, Applitools, and Jenkins.
+
+### Watir
+
+We chose to use [Watir](http://watir.com/) for the web automation. We have previously used it on our other [Sittercity](https://www.sittercity.com/) automation suites. It is easy to learn and very powerful in the right hands.
+
+### Cucumber
+Cucumber has many benefits for us. Since it's how we strive to structure our acceptance criteria. ♥ Cucumber and Watir-WebDriver sitting in a tree ♥ These two programs work really well with each other. It allows us to automate our web app and at the same time make the features easily readable to stakeholders, without any layers of abstraction.
```

Comment:
I personally would move the 'sitting in a tree' line after the 'they work well together line, like so:

```
Another big benefit is that Cucumber works really well with Watir. ♥ Cucumber and Watir-WebDriver sitting in a tree ♥ 
```

Maybe throw an emoji in there and start a new paragraph? Nothing saying that we can't have multiple paragraphs per section. I think it would read better to have this:

```
Cucumber (and the Gherkin DSL) has many benefits for us, chief among them being that it is how we strive to structure our acceptance criteria. Another big benefit is that Cucumber works really, really well with Watir. ♥ Cucumber and Watir-WebDriver sitting in a tree ♥  (some cool smile emoji)

This combination allows us to automate the testing of our web app and at the same time make the features easily readable to stakeholders, without any layers of abstraction. It's win-win-win for all involved.
```

What do you think of something like that?